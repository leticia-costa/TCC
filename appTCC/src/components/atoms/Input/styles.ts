import { StyleSheet } from "react-native";
import colors from "../../../utils/colors";

export const styles = StyleSheet.create({
  inputContainer: {
    width: "100%",
    height: 90,
    display: "flex",
    justifyContent: "center",
    paddingHorizontal: 10,
    marginVertical: 6,
    borderColor: colors.gray,
    borderRadius: 15,
    borderWidth: 2,
  },
  input: {
    fontSize: 18,
  },
});
