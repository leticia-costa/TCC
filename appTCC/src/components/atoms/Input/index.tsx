import React from 'react';
import { View, TextInput, KeyboardTypeOptions, StyleProp, TextStyle, ReturnKeyTypeOptions } from 'react-native';
import { styles } from './styles';

interface InputProps {
  id?: string
  placeholder?: string;
  value?: string;
  onChangeText?: (text: string) => void;
  style?: StyleProp<TextStyle>;
  type: KeyboardTypeOptions | undefined
  onSubmitEditing?: () => void
  ref?: React.LegacyRef<TextInput> 
  returnKeyType?: ReturnKeyTypeOptions
}

const Input: React.FC<InputProps> = ({ placeholder, returnKeyType, value, id, onChangeText, style, type, onSubmitEditing, ref }) => {
  return (
    <View style={[styles.inputContainer, style]}>
      <TextInput
        id={id}
        ref={ref}
        returnKeyType={returnKeyType}
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        style={[styles.input, style]}
        keyboardType={type}
        onSubmitEditing={onSubmitEditing}
      />
    </View>
  );
};

export default Input;
