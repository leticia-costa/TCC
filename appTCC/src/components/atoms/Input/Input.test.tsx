import React from "react";
import renderer from "react-test-renderer";
import Input from ".";

it("renders correctly", () => {
  const tree = renderer.create(<Input type="default" />).toJSON();
  expect(tree).toMatchSnapshot();
});
