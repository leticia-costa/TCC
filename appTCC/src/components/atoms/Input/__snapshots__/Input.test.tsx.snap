// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "borderColor": "#828B99",
        "borderRadius": 15,
        "borderWidth": 2,
        "display": "flex",
        "height": 90,
        "justifyContent": "center",
        "marginVertical": 6,
        "paddingHorizontal": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <TextInput
    keyboardType="default"
    style={
      [
        {
          "fontSize": 18,
        },
        undefined,
      ]
    }
  />
</View>
`;
